import "sofa/pbrpc/rpc_option.proto";
package mdt.LogSchedulerService;

message EchoRequest {
    required string message = 1;
}
message EchoResponse {
    required string message = 1;
}

message RegisterNodeRequest {
    optional string server_addr = 1; // ip:port 
}

message RegisterNodeResponse {
    optional int32 error_code = 1;
}

message GetNodeListRequest {
    optional string current_server_addr = 1; 
}

message GetNodeListResponse {
    repeated string server_list = 1;
    optional string primary_server_addr = 2;
}

// rpc service
service LogSchedulerService {
    rpc Echo(EchoRequest) returns(EchoResponse);
    rpc RegisterNode(RegisterNodeRequest) returns(RegisterNodeResponse);
    rpc GetNodeList(GetNodeListRequest) returns(GetNodeListResponse);
}
option py_generic_services = true;
option cc_generic_services = true;


